cin_validator/ingress.py

```diff
@@ -330,14 +341,11 @@ def create_Assessments(self, cin_detail):
             if assessment_factors:
                 # if statement handles the non-iterable NoneType that .find produces if the element is not present.
                 for factor in assessment_factors:
-                    assessment_dict = {
-                        "LAchildID": self.LAchildID,
-                        "CINdetailsID": self.CINdetailsID,
-                    }
-                    assessment_dict = get_values(elements, assessment_dict, assessment)
-                    # the get_values function will not find AssessmentFactors on that level so it'll assign it to NaN
-                    assessment_dict["AssessmentFactors"] = factor.text
-                    assessments_list.append(assessment_dict)
+                    assessment_block_maker(factor)
+            # else needed to build blocks in instances where assessments aren't completed which means there's no assessment factors to build with.
+            else:
+                for assessment in assessments:
+                    assessment_block_maker(assessment)
```

Comment:
I'm here. Giving this some thought.
Since this else statement is within the a for-assessment loop already, it shouldn't have to loop through anything when assessment factors are not found and should instead behave similar to a create_section47 ingress process.

Putting a pin on it. Will come back after the release to refactor.